examples/bell-variational/functions.py

```diff
@@ -0,0 +1,84 @@
+import numpy as np
+from qibo import gates
+from qibo.models import Circuit
+
+
+def bell_circuit(basis):
+    '''Create a Bell circuit with a distinct measurement basis and parametrizable gates.
+    Args:
+        basis (str): '00', '01, '10', '11' where a '1' marks a measurement in the X basis 
+                    and a '0' a measurement in the Z basis.
+    
+    Returns:
+        :class:`qibo.core.circuit.Circuit`
+    
+    '''
+    c = Circuit(2)
+    c.add(gates.RY(0, theta=0))
+    c.add(gates.CNOT(0, 1))
+    c.add(gates.RY(0, theta=0))
+    for a in range(2):
+        if basis[a] == '1':
+            c.add(gates.H(a))
+    c.add(gates.M(*range(2)))
+    return c
+
+
+def set_parametrized_circuits():
+    '''Create all Bell circuits needed to compute the CHSH inequelities.
+    Returns:
+        list(:class:`qibo.core.circuit.Circuit`)
+    
+    '''
+    chsh_circuits = []
+    basis = ['00', '01', '10', '11']
+    for base in basis:
+        chsh_circuits.append(bell_circuit(base))
+    return chsh_circuits
+
+
+def compute_chsh(frequencies):
+    '''Computes the CHSH inequelity value given the restults of an experiment.
+    Args:
+        frequencies (list): list of dictionaries with the result for the measurement
+                            and the number of times it repeated, for each measurement 
+                            basis
+    
+    Returns:
+        chsh (real): value of the CHSH inequality.
+    
+    '''
+    chsh = 0
+    aux = 0
+    for freq in frequencies:
+        for outcome in freq:
+            if aux == 1:
+                chsh -= (-1)**(int(outcome[0])+int(outcome[1]))*freq[outcome]
+            else:
+                chsh += (-1)**(int(outcome[0])+int(outcome[1]))*freq[outcome]
+        aux += 1
+    nshots = sum(freq[x] for x in freq)
```

Comment:
This line looks suspicious. This is using the last element of frequencies, freq is undefined in this context.

Reply:
It was a way to compute the nshots without giving it to the function, but now added as a parameter.